website/docs/about_code_generation.mdx

```diff
@@ -55,13 +79,133 @@ This includes but is not limited to:
 - some Riverpod features will be available only with code generation.
 
 At the same time, many applications already use code generation with packages such
-as [Freezed](https://pub.dev/packages/freezed) or [json_serializable](https://pub.dev/packages/json_serializable).  
+as [Freezed](https://pub.dev/packages/freezed) or [json_serializable](https://pub.dev/packages/json_serializable).
 In that case, your project probably is already setup for code generation, and
 using for Riverpod should be simple.
 
+## Provider type
+
+### <span style={RED_STYLE}>[Sync/Async]</span><span style={BLUE_STYLE}>[Stateless/Stateful]</span>Provider
```

Comment:
Although very pretty, I wouldn't use this as the section name.   
Folks coming to this page don't know what this means.

It rather should be explained within the content of this section.

Instead I'd use "defining a provider" as section name.